core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -321,9 +321,13 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
   override def getLastUpdatedTime(): Long = lastScanTime.get()
 
   override def getAppUI(appId: String, attemptId: Option[String]): Option[LoadedAppUI] = {
+    val logPath = RollingEventLogFilesWriter.EVENT_LOG_DIR_NAME_PREFIX +
+        Utils.nameForAppAndAttempt(appId, attemptId)
     val app = try {
       load(appId)
      } catch {
+      case _: NoSuchElementException if this.conf.get(EVENT_LOG_ROLLING_ON_DEMAND_LOAD_ENABLED) =>
+        loadFromFallbackLocation(appId, attemptId, logPath)
```

Comment:
What if `EVENT_LOG_ENABLE_ROLLING` is disabled? Should we only do this if `EVENT_LOG_ENABLE_ROLLING` is enabled?

Reply:
Thanks. I'll update like that.

Reply:
Initially, I proposed the config name `spark.history.fs.update.onDemandEnabled` because this is SHS setting. However, it's changed to during the review in order to be clear in the context of `rolling`.

```
- spark.history.fs.update.onDemandEnabled
+ spark.history.fs.eventLog.rolling.onDemandLoadEnabled
```

Reply:
It sounds like confusing. Basically, it's the same for event log compression codec. 

Since a Spark job can choose `spark.eventLog.compress` and `spark.eventLog.compression.codec` arbitrarily, we need to inference from the file name. It's inevitable because Writers(Spark job) and Reader(SHS) are independent.